cmd/query/app/apiv3/grpc_handler.go

```diff
@@ -74,17 +80,17 @@ func (h *Handler) FindTraces(request *api_v3.FindTracesRequest, stream api_v3.Qu
 		queryParams.DurationMax = d
 	}
 
-	traces, err := h.QueryService.FindTraces(stream.Context(), queryParams)
+	traces, err := h.QueryService.FindTraces(ctx, queryParams)
 	if err != nil {
 		return err
 	}
 	for _, t := range traces {
-		td, err := modelToOTLP(t.GetSpans())
-		if err != nil {
-			return err
-		}
+		td := modelToOTLP(t.GetSpans())
 		tracesData := api_v3.TracesData(td)
-		stream.Send(&tracesData)
+		if err := streamSend(&tracesData); err != nil {
+			return status.Error(codes.Internal,
+				fmt.Sprintf("failed to send response stream chunk to client: %v", err))
```

Comment:
should this be `err.Error()` to get the string representation of the error?

Reply:
I think `%v` will do that anyway for the error.